### PR TITLE
feat: Starknet "add_declare_transaction"

### PR DESCRIPTION
**Pull request by danilowhk**
- Added add_declare_transaction CLI command.
- Added a new endpoint to add a declare transaction on Starknet Network
- Implemented tests for Cli, Core and RestAPI
- Updated ethers library

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: #161 

# What is the new behavior?

-
-
-

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

# Other information

### CODE REVIEW

File: beerus_cli/src/model.rs [hunk 1]
```diff
@@ -239,6 +241,26 @@ pub enum StarkNetSubCommands {
         block_id: String,
     },
     QuerySyncing {},
+    AddDeclareTransaction {
+        /// Max fee
+        #[arg(short, long, value_name = "MAX_FEE")]
+        max_fee: String,
+        /// Max fee
```
Comment (LucasLvy) [inline beerus_cli/src/model.rs:248]:
```suggestion
        /// Declare tx version
```

File: beerus_cli/src/starknet/mod.rs [hunk 2]
```diff
@@ -293,3 +293,54 @@ pub async fn query_starknet_syncing(beerus: BeerusLightClient) -> Result<Command
...
     ))
 }
+
+/// Add an Declare transaction to the StarkNet network.
+/// # Arguments
+/// * `beerus` - The Beerus light client.
+/// * `max_fee` - The maximum fee.
+/// * `signature` - The signature.
+/// * `nonce` - The nonce.
+/// * `contract_address` - The contract address.
```
Comment (LucasLvy) [inline beerus_cli/src/starknet/mod.rs:303]:
```suggestion
/// * `contract_class` - The contract class.
```

File: beerus_cli/src/model.rs [hunk 3]
```diff
@@ -239,6 +241,26 @@ pub enum StarkNetSubCommands {
...
+        #[arg(short, long, value_name = "VERSION")]
+        version: String,
+        /// The signature
+        #[arg(short, long, value_name = "SIGNATURE", value_delimiter = ',')]
+        signature: Vec<String>,
+        /// The nonce
+        #[arg(short, long, value_name = "NONCE")]
+        nonce: String,
+        /// The contract address
```
Comment (LucasLvy) [inline beerus_cli/src/model.rs:257]:
```suggestion
        /// The contract class 
```
Which is the full contract right ?

**Review by LucasLvy — DISMISSED** [review]

File: beerus_cli/src/model.rs (hunk 3 above)
Comment (danilowhk) [reply beerus_cli/src/model.rs:257]: Yes! Correct, hehe thx for fixing the comments!

**Review by drspacemn — APPROVED** [review]
lgtm

**Pull request merged**